### PR TITLE
[TE6] Read Ethernet Network Diagnostic attributes from platform at run-time

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* Currently, the Ethernet Network Diagnostic attributes are fully managed by the attribute store, we can only specify the default values in its xml definitions. We need to populate those attributes with the actual value from platform in run-time.

#### Change overview
* Change Ethernet Network Diagnostic to 'External' in lighting-app.
* Read Ethernet Network Diagnostic attributes from platform at run-time.

#### Testing
How was this tested? (at least one bullet point required)
* ./chip-tool ethernetnetworkdiagnostics read packet-rx-count 0

```
[1631236896.763861][1715471:1715478] CHIP:EM: Removed CHIP MsgId:00000000 from RetransTable
[1631236896.763876][1715471:1715478] CHIP:DMG: ReportData =
[1631236896.763883][1715471:1715478] CHIP:DMG: {
[1631236896.763888][1715471:1715478] CHIP:DMG: 	AttributeDataList =
[1631236896.763894][1715471:1715478] CHIP:DMG: 	[
[1631236896.763902][1715471:1715478] CHIP:DMG: 		AttributeDataElement =
[1631236896.763910][1715471:1715478] CHIP:DMG: 		{
[1631236896.763917][1715471:1715478] CHIP:DMG: 			AttributePath =
[1631236896.763924][1715471:1715478] CHIP:DMG: 			{
[1631236896.763932][1715471:1715478] CHIP:DMG: 				NodeId = 0x47a899eac0fe5bfa,
[1631236896.763941][1715471:1715478] CHIP:DMG: 				EndpointId = 0x0,
[1631236896.763950][1715471:1715478] CHIP:DMG: 				ClusterId = 0x37,
[1631236896.763958][1715471:1715478] CHIP:DMG: 				FieldTag = 0x2,
[1631236896.763966][1715471:1715478] CHIP:DMG: 			}
[1631236896.763976][1715471:1715478] CHIP:DMG: 				
[1631236896.763985][1715471:1715478] CHIP:DMG: 			Data = 4346, 
[1631236896.763992][1715471:1715478] CHIP:DMG: 			DataElementVersion = 0x0,
[1631236896.764000][1715471:1715478] CHIP:DMG: 		},
[1631236896.764010][1715471:1715478] CHIP:DMG: 		
[1631236896.764017][1715471:1715478] CHIP:DMG: 	],
[1631236896.764027][1715471:1715478] CHIP:DMG: 	
[1631236896.764033][1715471:1715478] CHIP:DMG: }
[1631236896.764042][1715471:1715478] CHIP:DMG: ProcessReportData handles the initial report
[1631236896.764069][1715471:1715478] CHIP:ZCL: ReadAttributesResponse:
[1631236896.764075][1715471:1715478] CHIP:ZCL:   ClusterId: 0x0000_0037
[1631236896.764083][1715471:1715478] CHIP:ZCL:   attributeId: 0x0000_0002
[1631236896.764089][1715471:1715478] CHIP:ZCL:   status: Success                (0x0000)
[1631236896.764096][1715471:1715478] CHIP:ZCL:   attribute TLV Type: 0x04
[1631236896.764103][1715471:1715478] CHIP:TOO: Int64u attribute Response: 4346
```

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]
Approved, however with  the note that needing to copy&paste the implementation between all-clusters-app and lighting-app makes me sad:

- other examples will NOT have this
- this is a 'base cluster' so I would rather expect our code to have this working with as simple as some 'enableEthernetDiagnostics' call in main in an example app. Copy&pasting of 100 lines of code should not be required.

**Review by woody-apple — APPROVED** [review]

**Comment by yufengwangca** [comment]
> Approved, however with the note that needing to copy&paste the implementation between all-clusters-app and lighting-app makes me sad:
> 
> * other examples will NOT have this
> * this is a 'base cluster' so I would rather expect our code to have this working with as simple as some 'enableEthernetDiagnostics' call in main in an example app. Copy&pasting of 100 lines of code should not be required.

These implementations are platform specific code and currently different examples do not share the code for this part. We do have a lot of dup in this area, need to figure out a common solution to address all of them all. We may need to reactor the current example architecture to achieve this goal.

File: src/include/platform/CHIPDeviceConfig.h [hunk 1]
```diff
@@ -826,6 +826,18 @@
...
+/**
+ * @def CHIP_DEVICE_CONFIG_ETHERNET_IF_NAME
+ *
+ * @brief
+ *   Default ethernet network interface name used to centralize all metrics that are
+ *   relevant to a potential Ethernet connection to a Node.
+ *
+ */
+#ifndef CHIP_DEVICE_CONFIG_ETHERNET_IF_NAME
+#define CHIP_DEVICE_CONFIG_ETHERNET_IF_NAME "enp0s25"
```
Comment (msandstedt) [inline src/include/platform/CHIPDeviceConfig.h:838]: How can we possibly have any sane default interface name in this shared config file?  Isn't this by definition platform-specific?

Comment (yufengwangca) [reply src/include/platform/CHIPDeviceConfig.h:838]: This is the default config value for platform layer, each platform can overwrite this config within their own CHIPDeviceBuildConfig.h

Comment (yufengwangca) [reply src/include/platform/CHIPDeviceConfig.h:838]: Maybe we can remove the generic reference name and move its define to CHIPDeviceBuildConfig.h directly.

Comment (yufengwangca) [reply src/include/platform/CHIPDeviceConfig.h:838]: Addressed in https://github.com/project-chip/connectedhomeip/pull/9647

**Pull request merged**